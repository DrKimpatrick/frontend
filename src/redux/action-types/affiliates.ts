export enum AffiliateTypes {
  AddMoreInfo = 'Affiliate/AddMoreInfo',
  Errors = 'Affiliate/Errors',
  Loading = 'Affiliate/Loading'
}

// export interface CompanyOrSchool {
//   _id?: string;
//   name: string;
//   address: string;
//   website: string;
//   accountManagerName: string;
//   email: string;
//   phone: string;
//   userId?: string;
//   createdAt?: string;
//   updatedAt?: string;
// }

interface AddMoreInfo {
  type: typeof AffiliateTypes.AddMoreInfo;
  payload: {
    data: any;
  };
}

interface Errors {
  type: typeof AffiliateTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof AffiliateTypes.Loading;
}

export type CompanyActionTypes = AddMoreInfo | Errors | Loading;
