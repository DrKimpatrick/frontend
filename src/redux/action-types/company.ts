export enum CompanyTypes {
  AddCompany = 'Company/AddCompany',
  GetSpecificCompany = 'Company/GetSpecificCompany',
  UpdateCompany = 'Company/UpdateCompany',
  DeleteCompany = 'Company/DeleteCompany',
  Errors = 'Company/Errors',
  Loading = 'Company/Loading'
}

export interface CompanyOrSchool {
  _id?: string;
  name: string;
  address: string;
  website: string;
  accountManagerName: string;
  email: string;
  phone: string;
  userId?: string;
  createdAt?: string;
  updatedAt?: string;
}

interface AddCompany {
  type: typeof CompanyTypes.AddCompany;
  payload: {
    data: CompanyOrSchool;
  };
}

interface Errors {
  type: typeof CompanyTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof CompanyTypes.Loading;
}

export type CompanyActionTypes = AddCompany | Errors | Loading;
