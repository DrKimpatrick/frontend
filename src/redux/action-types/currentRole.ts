export const CURRENT_ROLE_LOADING = 'CURRENT_ROLE_LOADING';
export const CURRENT_ROLE_FAIL = 'CURRENT_ROLE_FAIL';
export const CURRENT_ROLE_SUCCESS = 'FORGOT_SUCCESS';

export interface CurrentRole {
    email: string
}

export interface currentRoleLoading  {
    type: typeof CURRENT_ROLE_LOADING
}

export interface currentRoleFail  {
    type: typeof CURRENT_ROLE_FAIL,
    payload: string
}

export interface currentRoleSuccess  {
    type: typeof CURRENT_ROLE_SUCCESS,
    payload: any
}

export type CurrentRoleDispatchTypes = currentRoleLoading | currentRoleFail | currentRoleSuccess;
