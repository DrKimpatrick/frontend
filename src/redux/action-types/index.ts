// This is where we will add action types