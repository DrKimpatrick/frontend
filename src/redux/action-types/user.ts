import { Employment } from './employment';
import { Education } from './education';
import { Skill } from './skill';

export enum UserTypes {
  CurrentUser = 'User/CurrentUser',
  Errors = 'User/Errors',
  ListSpecificUser = 'User/ListSpecificUser',
  Loading = 'User/Loading',
  ListUserSkill = 'User/ListUserSkill',
  ListUserByRole = 'User/ListUserByRole',
  ListTalentUser = 'User/ListTalentUser',
  ListHumanResourceUser = 'User/ListHumanResourceUser',
  ListAffiliateUser = 'User/ListAffiliateUser',
  ListHrAdminUser = 'User/ListHrAdminUser',
  ListTrainingUser = 'User/ListTrainingUser',
  ListRecruiterUser = 'User/ListRecruiterUser',
  ListCompanyUser = 'User/ListCompany',
  ActivePath = 'User/ActivePath',
  UpdateUser = 'User/UpdateUser',
  UserEducationLoading = 'User/UserEducationLoading',
  UserEmploymentLoading = 'User/UserEmploymentLoading',
  UserSkillLoading = 'User/UserSkillLoading',
  SetProfileProcess = 'User/SetProfileProcess'
}

export enum VerificationStatus {
  Verified = 'verified',
  Unverified = 'unverified',
  InProgress = 'inProgress'
}

export enum UserRole {
  Talent = 'talent',
  EducationUser = 'education',
  SuperAdmin = 'super_admin',
  RecruitmentAdmin = 'recruitment_admin',
  HrAdmin = 'hr_admin',
  CompanyAdmin = 'company_admin',
  TrainingAdmin = 'training_admin',
  TrainingAffiliate = 'training_affiliate'
}

export enum TalentProcess {
  CurrentRole = 'currentRole',
  SkillRanking = 'skillRanking',
  RecentEmployer = 'recentEmployer',
  AddEducation = 'addEducation',
  Completed = 'completed',
  SingleEducation = 'singleEducation',
  ListEducation = 'listEducation',
  SingleEmployment = 'singleEmployment',
  ListEmployment = 'listEmployment'
}

export enum AdminsProcess {
  AddCompany = 'AddCompany',
  AddSchool = 'AddSchool',
  AddPlan = 'AddPlan',
  Payment = 'Payment',
  Completed = 'Completed'
}

export interface User {
  _id: string;
  username: string;
  email: string;
  roles: string[];
  updatedAt: string;
  dateRegistered: string;
  paymentStatus: string;
  featureChoice: string;
  verified: boolean;
  employmentHistory?: Employment[];
  educationHistory?: Education[];
  profileProcess?: string;
  profilePicture?: string;
  bio?: string;
  stripeSubscriptionId?: string;
}

export interface UserSkill {
  _id: string;
  verificationStatus: string;
  level: string;
  skill: Skill;
  createdAt?: string;
  updatedAt?: string;
}

interface CurrentUser {
  type: typeof UserTypes.CurrentUser;
  payload: {
    data: User;
  };
}

interface Errors {
  type: typeof UserTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof UserTypes.Loading;
  payload: {
    loading: boolean;
  };
}

interface ListSpecificUser {
  type: typeof UserTypes.ListSpecificUser;
  payload: {
    data: User;
  };
}

interface ListUserSkill {
  type: typeof UserTypes.ListUserSkill;
  payload: {
    data: UserSkill[];
  };
}

interface ListUserByRole {
  type: typeof UserTypes.ListUserByRole;
  payload: {
    data: {
      totalDocs: number;
      data: User[];
      currentPage: number;
    };
  };
}

interface ActivePath {
  type: typeof UserTypes.ActivePath;
  payload: {
    data: string;
  };
}

interface UpdateUser {
  type: typeof UserTypes.UpdateUser;
  payload: {
    data: User;
  };
}

interface ListTalentUser {
  type: typeof UserTypes.ListTalentUser;
  payload: {
    data: {
      totalDocs: number;
      data: User[];
      currentPage: number;
    };
  };
}

interface ListHrAdminUser {
  type: typeof UserTypes.ListHrAdminUser;
  payload: {
    data: {
      totalDocs: number;
      data: User[];
      currentPage: number;
    };
  };
}

interface ListRecruiterUser {
  type: typeof UserTypes.ListRecruiterUser;
  payload: {
    data: {
      totalDocs: number;
      data: User[];
      currentPage: number;
    };
  };
}

interface ListCompanyUser {
  type: typeof UserTypes.ListCompanyUser;
  payload: {
    data: {
      totalDocs: number;
      data: User[];
      currentPage: number;
    };
  };
}

interface ListAffiliateUser {
  type: typeof UserTypes.ListAffiliateUser;
  payload: {
    data: {
      totalDocs: number;
      data: User[];
      currentPage: number;
    };
  };
}

interface ListTrainingAffiliateUser {
  type: typeof UserTypes.ListTrainingUser;
  payload: {
    data: {
      totalDocs: number;
      data: User[];
      currentPage: number;
    };
  };
}

interface UserEducationLoading {
  type: typeof UserTypes.UserEducationLoading;
  payload: {
    loading: boolean;
  };
}

interface UserEmploymentLoading {
  type: typeof UserTypes.UserEmploymentLoading;
  payload: {
    loading: boolean;
  };
}

interface UserSkillLoading {
  type: typeof UserTypes.UserSkillLoading;
  payload: {
    loading: boolean;
  };
}

interface SetProfileProcess {
  type: typeof UserTypes.SetProfileProcess;
  payload: {
    data: string;
  };
}
export type UserTypeActions =
  | CurrentUser
  | Errors
  | ListSpecificUser
  | Loading
  | ListUserSkill
  | ListUserByRole
  | ActivePath
  | UpdateUser
  | ListTalentUser
  | ListAffiliateUser
  | ListTrainingAffiliateUser
  | ListRecruiterUser
  | ListCompanyUser
  | ListHrAdminUser
  | UserEducationLoading
  | UserEmploymentLoading
  | UserSkillLoading
  | SetProfileProcess;
