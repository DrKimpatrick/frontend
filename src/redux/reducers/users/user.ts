import { SkillTypes, SkillActionTypes } from 'redux/action-types/skill';
import {
  RegisterDispatchTypes,
  REGISTER_LOADING,
  REGISTER_FAIL,
  REGISTER_SUCCESS
} from 'redux/action-types/getStarted';

import {
  LoginDispatchTypes,
  LOGIN_LOADING,
  LOGIN_FAIL,
  LOGIN_SUCCESS
} from 'redux/action-types/login';

import {
  FORGOT_FAIL,
  FORGOT_LOADING,
  FORGOT_SUCCESS,
  ForgotDispatchTypes
} from 'redux/action-types/forgotPass';

import {
  UserTypeActions,
  UserTypes,
  User,
  UserSkill
} from 'redux/action-types/user';

import {
  RESET_FAIL,
  RESET_LOADING,
  RESET_SUCCESS,
  ResetDispatchTypes
} from 'redux/action-types/resetPass';

import {
  VERIFY_SUCCESS,
  VERIFY_LOADING,
  VERIFY_FAIL,
  VerifyDispatchTypes
} from 'redux/action-types/verifyAcc';

interface InitialState {
  currentUser: {
    isLoggedIn: boolean;
    data: object | null | any;
    loading: boolean;
    error?: any;
    message?: string;
    created?: boolean;
  };
  user?: User;
  loading?: boolean;
  errors?: any;
  specificUser?: User;
  userSkill?: UserSkill[];
  usersByRole?: {
    totalDocs?: number;
    users?: User[];
    currentPage?: number;
  };
  // skills?: AddedSkill[];
  activePath: string | null;
}

const initialState: InitialState = {
  currentUser: {
    isLoggedIn: !!localStorage.getItem('token'),
    loading: false,
    created: false,
    data: {
      firstName: 'John',
      lastName: 'Smith',
      profilePicture:
        'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
    }
  },
  activePath: null
};

export const userReducer = (
  state = initialState,
  action:
    | RegisterDispatchTypes
    | LoginDispatchTypes
    | ForgotDispatchTypes
    | ResetDispatchTypes
    | VerifyDispatchTypes
    | UserTypeActions
    | SkillActionTypes
) => {
  switch (action.type) {
    case REGISTER_FAIL:
      return {
        ...state,
        errorSignup: action.payload,
        created: true,
        loading: false
      };

    case REGISTER_SUCCESS:
      return {
        ...state,
        loading: false,
        currentUser: {
          data: action.payload,
          isLoggedIn: false,
          created: true,
          error: null
        }
      };
    case LOGIN_LOADING:
      return {
        ...state,
        loading: true
      };
    case FORGOT_LOADING:
      return {
        ...state,
        loading: true
      };
    case RESET_LOADING:
      return {
        ...state,
        loading: true
      };
    case VERIFY_LOADING:
      return {
        ...state,
        loading: true
      };
    case REGISTER_LOADING:
      return {
        ...state,
        loading: true
      };
    case LOGIN_FAIL:
      return {
        ...state,
        errorLogin: action.payload,
        loading: false
      };

    case LOGIN_SUCCESS:
      return {
        ...state,
        currentUser: action.payload,
        loading: false
      };

    case FORGOT_FAIL:
      return {
        ...state,
        errorForgot: action.payload,
        loading: false
      };

    case FORGOT_SUCCESS:
      return {
        ...state,
        errorForgot: null,
        loading: false
      };

    case RESET_FAIL:
      return {
        ...state,
        errorReset: action.payload,
        loading: false
      };

    case RESET_SUCCESS:
      return {
        ...state,
        errorReset: null,
        resetSuccess: true,
        loading: false
      };

    case VERIFY_FAIL:
      return {
        ...state,
        errorVerify: action.payload,
        loading: false
      };

    case VERIFY_SUCCESS:
      return {
        ...state,
        message: action.payload,
        errorVerify: null,
        loading: false
      };

    case UserTypes.CurrentUser:
      return { ...state, user: action.payload.data, loading: false };

    case UserTypes.Errors:
      return { ...state, errors: action.payload.errors, loading: false };

    case UserTypes.Loading:
      return { ...state, loading: action.payload.loading, errors: null };

    case UserTypes.ListSpecificUser:
      return {
        ...state,
        specificUser: action.payload.data,
        loading: false,
        errors: undefined
      };

    case UserTypes.ListUserSkill:
      return { ...state, userSkill: action.payload.data, loading: false };

    case SkillTypes.GetSkillsFail:
      return { ...state, errors: action.payload, loading: false };

    case SkillTypes.GetSkillsLoading:
      return { ...state, loading: true };

    case SkillTypes.GetSkillsSuccess:
      return { ...state, skills: action.payload, loading: false, errors: null };

    case SkillTypes.SkillFail:
      return { ...state, errors: action.payload, loading: false };

    case SkillTypes.SkillSuccess:
      return {
        ...state,
        addedSkill: action.payload,
        loading: false,
        errors: null
      };

    case SkillTypes.SkillLoading:
      return { ...state, loading: true };

    case UserTypes.ListUserByRole:
      return {
        ...state,
        usersByRole: {
          users: action.payload.data.data,
          totalDocs: action.payload.data.totalDocs,
          currentPage: action.payload.data.currentPage
        },
        loading: false
      };

    case UserTypes.ActivePath:
      return { ...state, activePath: action.payload.data };

    case UserTypes.UpdateUser:
      return {
        ...state,
        user: action.payload.data,
        loading: false,
        errors: null
      };

    default:
      return state;
  }
};

export default userReducer;
