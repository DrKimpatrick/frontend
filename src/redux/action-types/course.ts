import { CourseOwnerType, Course, UserCourseType } from 'redux/actions/course';
import { User } from 'redux/actions/user';

export enum CourseTypes {
  AddCourse = 'CourseTypes/AddCourse',
  UpdateCourse = 'CourseTypes/UpdateCourse',
  ListPendingCourse = 'CourseTypes/ListPendingCourse',
  ListAcceptedCourse = 'CourseTypes/ListAcceptedCourse',
  ListDeclinedCourse = 'CourseTypes/ListDeclinedCourse',
  Loading = 'CourseTypes/Loading',
  Errors = 'CourseTypes/Errors',
  NumberOfPendingCourse = 'CourseTypes/NumberOfPendingCourse',
  NumberOfDeclinedCourse = 'CourseTypes/NumberOfDeclinedCourse',
  NumberOfAcceptedCourse = 'CourseTypes/NumberOfAcceptedCourse',
  ListCourseDetail = 'CourseTypes/ListCourseDetail',
  CourseDetailLoading = 'CourseTypes/CourseDetailLoading',
  ListCourse = 'CourseTypes/ListCourse',
  ListCourseLoading = 'CourseTypes/ListCourseLoading',
  SubmitLoading = 'CourseTypes/SubmitLoading',
  ListCourseOwner = 'CourseTypes/ListCourseOwner',
  CourseOwnerLoading = 'CourseTypes/CourseOwnerLoading',
  UserCourse = 'CourseTypes/UserCourse',
  UserCourseLoading = 'CourseTypes/UserCourseLoading',
  UserWhoPaidCourse = 'CourseTypes/UserWhoPaidCourse',
  UserWhoPaidCourseLoading = 'CourseTypes/UserWhoPaidCourseLoading'
}

export enum CourseStatus {
  Accepted = 'accepted',
  Pending = 'pending',
  Declined = 'declined'
}

export enum CourseTimeFormat {
  Minute = 'Min',
  Hour = 'Hr'
}

export enum CourseLevel {
  Advanced = 'advanced',
  Beginner = 'beginner',
  Intermediate = 'intermediate'
}

export enum CourseBillingOptions {
  ONE_TIME = 'one-time',
  DAILY = 'day',
  WEEKLY = 'week',
  MONTHLY = 'month',
  YEARLY = 'year'
}

export interface CoursePagination {
  offset?: number;
  limit?: number;
}
interface ListAcceptedCourse {
  type: typeof CourseTypes.ListAcceptedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListPendingCourse {
  type: typeof CourseTypes.ListPendingCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListDeclinedCourse {
  type: typeof CourseTypes.ListDeclinedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}
interface Loading {
  type: typeof CourseTypes.Loading;
  payload: {
    loading: boolean;
  };
}

interface Errors {
  type: typeof CourseTypes.Errors;
  payload: {
    errors: any;
  };
}

interface NumberOfPendingCourse {
  type: typeof CourseTypes.NumberOfPendingCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface NumberOfDeclinedCourse {
  type: typeof CourseTypes.NumberOfDeclinedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface NumberOfAcceptedCourse {
  type: typeof CourseTypes.NumberOfAcceptedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListCourseDetail {
  type: typeof CourseTypes.ListCourseDetail;
  payload: {
    data: Course;
  };
}

interface CourseDetailLoading {
  type: typeof CourseTypes.CourseDetailLoading;
  payload: {
    loading: boolean;
  };
}

interface ListCourse {
  type: typeof CourseTypes.ListCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
    };
  };
}

interface ListCourseLoading {
  type: typeof CourseTypes.ListCourseLoading;
  payload: {
    loading: boolean;
  };
}

interface AddCourse {
  type: typeof CourseTypes.AddCourse;
  payload: {
    data: boolean;
  };
}

interface UpdateCourse {
  type: typeof CourseTypes.UpdateCourse;
  payload: {
    data: boolean;
  };
}

interface SubmitLoading {
  type: typeof CourseTypes.SubmitLoading;
  payload: {
    loading: boolean;
  };
}

interface ListCourseOwner {
  type: typeof CourseTypes.ListCourseOwner;
  payload: {
    data: CourseOwnerType;
  };
}

interface CourseOwnerLoading {
  type: typeof CourseTypes.CourseOwnerLoading;
  payload: {
    loading: boolean;
  };
}

interface UserCourse {
  type: typeof CourseTypes.UserCourse;
  payload: {
    data: UserCourseType;
  };
}

interface UserCourseLoading {
  type: typeof CourseTypes.UserCourseLoading;
  payload: {
    loading: boolean;
  };
}

interface UserWhoPaidCourse {
  type: typeof CourseTypes.UserWhoPaidCourse;
  payload: {
    data: User[];
  };
}

interface UserWhoPaidCourseLoading {
  type: typeof CourseTypes.UserWhoPaidCourseLoading;
  payload: {
    loading: boolean;
  };
}

export type CourseActionTypes =
  | ListAcceptedCourse
  | ListDeclinedCourse
  | ListPendingCourse
  | Loading
  | Errors
  | NumberOfAcceptedCourse
  | NumberOfDeclinedCourse
  | NumberOfPendingCourse
  | ListCourseDetail
  | CourseDetailLoading
  | ListCourse
  | ListCourseLoading
  | AddCourse
  | UpdateCourse
  | SubmitLoading
  | ListCourseOwner
  | CourseOwnerLoading
  | UserCourse
  | UserCourseLoading
  | UserWhoPaidCourse
  | UserWhoPaidCourseLoading;
