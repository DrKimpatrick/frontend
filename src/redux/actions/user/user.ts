import { Dispatch } from 'redux';
import { UserTypes } from 'redux/action-types/user';
import { AddSkill, SkillTypes } from 'redux/action-types/skill';
import ApiAction from '../../../helpers/apiAction';

export const listCurrentUser = () => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/users/me',
      method: 'GET',
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.CurrentUser,
          payload: { data: res.profile }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.Errors,
          payload: { errors: error.response.data }
        });
      }
    })
  );
};

export const listSpecificUser = (userId: string) => (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/users/${userId}`,
      method: 'GET',
      onStart: () => (dispatch: Dispatch) => {
        dispatch({ type: UserTypes.Loading, payload: { loading: true } });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.ListSpecificUser,
          payload: { data: res.profile }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.Errors,
          payload: { errors: error.response.data }
        });
      }
    })
  );
};

export const listUserSkill = (userId: string) => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: `/users/${userId}/skills`,
      method: 'GET',
      onStart: () => (dispatch: Dispatch) => {
        dispatch({ type: UserTypes.Loading, payload: { loading: true } });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.ListUserSkill,
          payload: { data: res.data }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.Errors,
          payload: { errors: error.response.data }
        });
      }
    })
  );
};

export const listAllSkill = () => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: `/skills`,
      method: 'GET',
      onStart: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.GetSkillsLoading,
          payload: { loading: true }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.GetSkillsSuccess,
          payload: res.data
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.GetSkillsFail,
          payload: { errors: error.response.data }
        });
      }
    })
  );
};

export const listUserByRole = (data: {
  page: number;
  limit: number;
  role: string;
}) => (dispatchAction: Dispatch) => {
  const { page, limit, role } = data;
  return dispatchAction(
    ApiAction({
      url: `/users/?roles=${role}&limit=${limit}&page=${page}`,
      method: 'GET',
      onStart: () => (dispatch: Dispatch) => {
        dispatch({ type: UserTypes.Loading, payload: { loading: true } });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.ListUserByRole,
          payload: { data: res }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.Errors,
          payload: { errors: error.response.data }
        });
      }
    })
  );
};

export const addedSkillsUser = (dataValues: AddSkill[]) => (
  dispatchAction: Dispatch
) => {
  console.log('data values', dataValues);
  return dispatchAction(
    ApiAction({
      url: '/skills/me',
      method: 'POST',
      data: dataValues,
      onStart: () => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.SkillLoading,
          payload: { loading: true }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.SkillSuccess,
          payload: { data: res }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: SkillTypes.SkillFail,
          payload: { errors: error.response.data }
        });
      }
    })
  );
};

export const setActivePath = (pathname: string) => (dispatch: Dispatch) => {
  dispatch({ type: UserTypes.ActivePath, payload: { data: pathname } });
};

export const updateUser = (data: any, userId: string) => (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/users/${userId}`,
      method: 'PATCH',
      data,
      onStart: () => (dispatch: Dispatch) => {
        dispatch({ type: UserTypes.Loading, payload: { loading: true } });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        delete res.profile.__v;
        setTimeout(() => {
          dispatch({
            type: UserTypes.UpdateUser,
            payload: { data: res.profile }
          });
        }, [1000]);
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: UserTypes.Errors,
          payload: { errors: error.response.data }
        });
      }
    })
  );
};
