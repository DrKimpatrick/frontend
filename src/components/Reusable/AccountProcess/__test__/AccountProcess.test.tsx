import React from 'react';
import Component from '../AccountProcess';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Store } from 'redux';
import thunk from 'redux-thunk';
import { TalentProcess } from 'redux/action-types/user';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';

let store: Store;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const initialState = {
  users: {
    user: {
      _id: 'id',
      username: 'username',
      email: 'email',
      roles: ['talent'],
      updatedAt: '2020-10-10',
      dateRegistered: '2020-10-10',
      paymentStatus: 'verified',
      featureChoice: 'local',
      verified: true,
      profileProcess: TalentProcess.Completed
    },
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    loading: false
  }
};

describe('Account Process', () => {
  afterEach(cleanup);

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={store}>
        <Router>
          <Component />
        </Router>
      </Provider>,
      div
    );
  });

  it('it should create snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Component />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  describe('should return splashScreen', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        users: {
          ...initialState.users,
          loading: true,
          user: { ...initialState.users.user, roles: [] }
        }
      });
    });
    it('it should return splashScreen when loading is true', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <Router>
              <Component />
            </Router>
          </Provider>
        )
        .toJSON();

      expect(tree).toBeTruthy();
    });
  });
});
