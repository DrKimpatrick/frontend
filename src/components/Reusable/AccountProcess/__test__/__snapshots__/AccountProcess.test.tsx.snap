// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Process it should create snapshot 1`] = `null`;
